aws/cloudformation/components/database.yml.erb

```diff
@@ -11,6 +11,19 @@
       Parameters: <%=values.compact.to_json%>
 <%end -%>
 
+<%  db_secrets = [nil, 'application-writer', 'readonly'].map do |user| -%>
+  DatabaseSecret<%=secret = user&.underscore&.camelcase%>:
+    Type: AWS::SecretsManager::Secret
+    Properties:
+      Description: !Sub "Secrets for accessing database<%user ? " user #{user}" : ''%> from ${AWS::StackName} CloudFormation stack"
+      GenerateSecretString:
+        SecretStringTemplate: !Sub '{"username": "<%=user || '${DatabaseUsername}'%>"}'
+        GenerateStringKey: password
+        PasswordLength: 10
+        ExcludePunctuation: True
+      Name: !Sub "CfnStack/${AWS::StackName}/database-<%=user || 'secret'%>"
+<%  secret; end -%>
+
 # We don't provision these in production via CloudFormation ... yet!
 <% unless rack_env?(:production)%>
```

Comment:
So the net change here is that you are moving these resources out of the 'unless prod' block to ensure they are deployed to prod too.